### PR TITLE
SAK-29813 New Webservices related with User. 

### PR DESCRIPTION
**Pull request by ddelblanco**
checkForUserById (in Sakai or external provider, by "User ID" , the actual Sakai one checkForUser is using the "eID")
isSuperUser (to know if it is or not)
resetAllUserWorkspace (to reset the workspaces, cleaning the database very deeply)

### CODE REVIEW

File: webservices/cxf/src/resources/applicationContext.xml [hunk 1]
```diff
@@ -160,7 +160,9 @@ http://cxf.apache.org/schemas/jaxws.xsd">
     <bean id="org.sakaiproject.webservices.SakaiJob" class="org.sakaiproject.webservices.SakaiJob" init-method="init" parent="org.sakaiproject.webservices.AbstractWebService" />
     <bean id="org.sakaiproject.webservices.SakaiLogin" class="org.sakaiproject.webservices.SakaiLogin" init-method="init" parent="org.sakaiproject.webservices.AbstractWebService" />
     <bean id="org.sakaiproject.webservices.SakaiPortalLogin" class="org.sakaiproject.webservices.SakaiPortalLogin" init-method="init" parent="org.sakaiproject.webservices.AbstractWebService" />
-    <bean id="org.sakaiproject.webservices.SakaiScript" class="org.sakaiproject.webservices.SakaiScript" init-method="init" parent="org.sakaiproject.webservices.AbstractWebService" />
+    <bean id="org.sakaiproject.webservices.SakaiScript" class="org.sakaiproject.webservices.SakaiScript" init-method="init" parent="org.sakaiproject.webservices.AbstractWebService" >
+        <property name="sqlService" ref="org.sakaiproject.db.api.SqlService"/>
+    </bean>
```
Comment (ottenhoff) [inline webservices/cxf/src/resources/applicationContext.xml]: This should be done above in the AbstractWebService

Comment (ddelblanco) [inline webservices/cxf/src/resources/applicationContext.xml]: Yes, I'll do, so If we add this in the Abstract and remove here, (as you say, it is the logical thing), we should remove that too in the SakaiReport one, that has the same addition.  I think that should be a different ticket anyway (I mean remove the reference to sqlService in the SakaiReport).

**Comment by steveswinsburg** [comment]
Again as per #1012, please refactor these web services into API methods. Touching database tables is not the concern of a web service and this must not be merged in its current form.

**Comment by ddelblanco** [comment]
I've created and alternative webservice deleting the sites. I'm not happy with this sentence:
List siteList = siteService.getSites(SelectionType.ANY, null, null,
                    null, SortType.NONE, null);
Because it selects all the sites in Sakai and then with a iterator I filter later here for the user sites.

But I've not found any way to retrieve only a list with the user workspaces. The param criteria seems not allow the "~" symbol as criteria to search in the siteId and pre-filter.

I don't know how big the sites list this can be in an instance with hundred thousands sites and users...

**Comment by steveswinsburg** [comment]
Best bet is to create a new SelectionType that filters what you need. IIRC this is backed by SQL so the filtering is done there.

**Comment by ddelblanco** [comment]
Well, with the actual parameters, SelectionType doesn't allows to create a filter to return only the user sites, so I've changed the siteService.getSites by siteService.getSiteIds, that returns a list of strings, so seems more efficient, including the case of a big number of sites.

**Comment by ddelblanco** [comment]
Seems that there are some conflicts. I'll solve that and resend the pull request.

**Pull request closed without merging**